src/js/components/agencyV2/visualizations/ObligationsByAwardType.jsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * ObligationsByAwardType.jsx
+ * Created by Brett Varney 4/08/21
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as d3 from 'd3';
+
+const propTypes = {
+    outer: PropTypes.arrayOf(
+        PropTypes.shape({
+            label: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
+            value: PropTypes.number.isRequired,
+            color: PropTypes.string.isRequired
+        })
+    ).isRequired,
+    inner: PropTypes.arrayOf(
+        PropTypes.shape({
+            label: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
+            value: PropTypes.number.isRequired,
+            color: PropTypes.string.isRequired
+        })
+    ).isRequired,
+    windowWidth: PropTypes.number.isRequired
+};
+
+export default function ObligationsByAwardType({ outer, inner, windowWidth }) {
+    const [chartRect, setChartRect] = React.useState([0, 0]); // height, width
+    const chartRef = React.useRef();
+
+    React.useEffect(() => {
+        const rect = chartRef.current.parentElement.getBoundingClientRect();
+        if (rect.height !== chartRect[0] || rect.width !== chartRect[1]) {
+            setChartRect([rect.height, rect.width]);
+        };
+    }, [windowWidth]);
+
+    const labelRadius = Math.min(chartRect[0], chartRect[1]) / 2;
+    const outerRadius = labelRadius * 0.7;
+    const outerStrokeWidth = 3;
+    const innerRadius = outerRadius - (outerStrokeWidth * 2);
+
+    // clear & append the svg object to the div
+    d3.select('#obl_chart').selectAll('*').remove();
```

Comment:
This is probably a larger discussion - but I noticed that when you change the FY picker, the chart does not update to reflect the new data. I think it's because d3 is rendering the chart instead of React. In our other d3 visualizations on USAS, we're using d3 to "do the math" and then manually drawing SVGs (which get updated on re-render).

Reply:
Interesting... it changes for me

Reply:
Ok I must have been looking at years with a very similar breakdown! This is great news that a more standard d3 approach works with `useRef` 🎉